src/coreclr/jit/codegenarm64.cpp

```diff
@@ -2764,6 +2758,18 @@ void CodeGen::genCodeForBinary(GenTreeOp* tree)
         genProduceReg(tree);
         return;
     }
+    else if (compiler->IsTargetAbi(CORINFO_NATIVEAOT_ABI) && TargetOS::IsWindows &&
+             op2->IsIconHandle(GTF_ICON_SECREL_OFFSET))
```

Comment:
So this constant isn't considered a reloc?

Reply:
It is and hence we add `EA_CNS_RELOC_FLG | EA_CNS_SEC_RELOC` flags below.